Create slurm cluster

YidingZhou opened the pull request.

singhkay commented on slurm/azuredeploy.json:
```diff
@@ -0,0 +1,309 @@
+                "description": "This is the location where the availability set will be deployed"
+            }
+        },
+        "MasterVMName": {
```
camelCase for these parameter names

singhkay commented on slurm/azuredeploy.json:
```diff
@@ -0,0 +1,309 @@
+                "description": "The machine name of the master node - not be confused with dnsName which is public visible name."
+            }
+        },
+        "WorkerVMName": {
```
camelCase for these parameter names

singhkay commented on slurm/azuredeploy.json:
Did you mean to use the nicname here?

singhkay commented on slurm/metadata.json:
```diff
@@ -0,0 +1,7 @@
+  "description": "Template spins up a SLURM cluster.", 
+  "githubUsername": "YidingZhou", 
+  "summary": "SLURM cluster",
+  "dateUpdated": "2015-05-18" 
```
Update this to a more recent date :)

singhkay commented:
Adding @johngossman for review as well

YidingZhou commented:
Updated date and camelCase.

johngossman commented on slurm/azuredeploy.json:
```diff
@@ -0,0 +1,309 @@
+                "description": "This is the location where the availability set will be deployed"
+            }
+        },
+        "masterVMName": {
```
Don't make this a parameter. Just hardcode "master"

johngossman commented on slurm/azuredeploy.json:
```diff
@@ -0,0 +1,309 @@
+                "description": "The machine name of the master node - not be confused with dnsName which is public visible name."
+            }
+        },
+        "workerVMName": {
```
Don't make this a parameter

johngossman commented on slurm/azuredeploy-parameters.json:
```diff
@@ -0,0 +1,11 @@
+{
```
This file should match the parameters in the template

johngossman commented on slurm/azuredeploy.json:
```diff
@@ -0,0 +1,309 @@
+                "description": "Unique DNS Name for the Storage Account where the Virtual Machine's disks will be placed."
+            }
+        },
+        "adminPassword": {
```
Put this parameter next to adminUser

johngossman commented on slurm/azuredeploy.json:
```diff
@@ -0,0 +1,309 @@
+                "description": "Admin password"
+            }
+        },
+        "vmSize": {
```
Provide allowedValues for this. Use D-sizes instead of A-sizes

johngossman commented:
Could you please squash your commits?

YidingZhou commented:
I've squashed the request. How do eliminate the acomghbot comment?

YidingZhou commented:
Squashed all commits and updated according to comments. #1 VM size, #2 Move VM name from parameters to variables, #3 Co-located username and password parameters, and #4 make parameter json match the deployment template.

singhkay commented on slurm/README.md:
```diff
@@ -0,0 +1,34 @@
+
+##Deploy
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FYidingZhou%2Fazure-quickstart-templates%2Fmaster%2Fslurm%2Fazuredeploy.json" target="_blank">
```
This needs to be updated to point to this repo

singhkay commented on slurm/azuredeploy.json:
```diff
@@ -0,0 +1,303 @@
+        "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('networkSettings').virtualNetworkName)]",
+        "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('networkSettings').subnet.dse.name)]",
+        "sshKeyPath": "[concat('/home/',parameters('adminUsername'),'/.ssh/authorized_keys')]",
+        "templateBaseUrl": "https://raw.githubusercontent.com/YidingZhou/azure-quickstart-templates/master/slurm/",
```
This needs to be updated for this repo as well

singhkay commented on slurm/azuredeploy.sh:
```diff
@@ -0,0 +1,126 @@
+#!/bin/sh
+
+# This script can be found on https://github.com/YidingZhou/azure-quickstart-templates/blob/master/slurm/azuredeploy.sh
```
This url needs update as well
